### PR TITLE
add helper to build a QgsRectangle from a WKT string

### PR DESCRIPTION
**Pull request by Gustry**
## Description

add a helper to build a QgsRectangle from a WKT string

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/core/geometry/qgsrectangle.cpp [hunk 1]
```diff
@@ -59,6 +60,24 @@ QgsRectangle::QgsRectangle( const QgsRectangle &r )
   mYmax = r.yMaximum();
 }
 
+QgsRectangle QgsRectangle::fromWkt( const QString &wkt )
+{
+  QgsGeometry geom = QgsGeometry::fromWkt( wkt );
+  if ( ! geom.isGeosValid() || geom.isMultipart() )
```
Comment (nyalldawson) [inline src/core/geometry/qgsrectangle.cpp]: Given that isGeosValid is the most expensive check, I'd move this one last (after all the cheap type/size checks)

File: src/core/geometry/qgsrectangle.h [hunk 2]
```diff
@@ -54,6 +54,13 @@ class CORE_EXPORT QgsRectangle
     ~QgsRectangle() = default;
 
     /**
+    * Creates a new rectangle from a WKT string.
+    * The WKT must contain only 4 vertices, representing a rectangle.
```
Comment (nyalldawson) [inline src/core/geometry/qgsrectangle.h]: Shouldn't this be 5 vertices?

Comment (nyalldawson) [reply src/core/geometry/qgsrectangle.h]: Better also mention here that only rectangles aligned with the X/y axis are supported.

**Comment by Gustry** [comment]
Thanks for the prompt review.
QgsRectangle is always aligned with X and Y axes, so it should be OK like this.

**Comment by nyalldawson** [comment]
> QgsRectangle is always aligned with X and Y axes, so it should be OK like this.

We both know this, but I think the docs should be explicit about behaviour like this for those less familiar with the qgis API ;)

**Comment by nyalldawson** [comment]
Thanks!

**Comment by NathanW2** [comment]
Oh very nice.  I had a use case for this the other day./

**Pull request merged**